### PR TITLE
Add sign-tx tests to prevent regression

### PR DESCRIPTION
**Pull request by gabrielKerekes**

### CODE REVIEW

File: tests/device_tests/test_msg_cardano_sign_transaction.py [hunk 1]
```diff
@@ -17,215 +17,274 @@
...
+NETWORK_IDS = {"mainnet": 0}
+
+SHELLEY_TEST_VECTORS_MNEMONIC = (
+    "test walk nut penalty hip pave soap entry language right filter choice"
+)
+
+# input flow sequence constants
+SWIPE = "SWIPE"
+YES = "YES"
+YIELD = "YIELD"
```
Comment (janmazak) [inline tests/device_tests/test_msg_cardano_sign_transaction.py:30]: This warrants a more detailed description. And perhaps grouping into a class with a descriptive yet short name? Do these constants correspond to some user actions? If yes, then UserAction would be a suitable name.

File: tests/device_tests/test_msg_cardano_sign_transaction.py [hunk 2]
```diff
@@ -17,215 +17,274 @@
...
+    expected_responses.append(messages.CardanoSignedTx())
 
     def input_flow():
-        yield
-        client.debug.swipe_up()
-        client.debug.press_yes()
-        yield
-        client.debug.swipe_up()
-        client.debug.press_yes()
+        for sub_sequence in input_flow_sequence:
```
Comment (janmazak) [inline tests/device_tests/test_msg_cardano_sign_transaction.py:275]: I don't understand this. A sequence is composed of members / elements / terms, not subsequences. The names are not self-explanatory.

**Comment by gabrielKerekes** [comment]
I tried to refactor it. Let me know if it's better. If not, we should maybe get on a quick call and come up with a better way together.

File: tests/device_tests/test_msg_cardano_sign_transaction.py [hunk 3]
```diff
@@ -17,215 +17,278 @@
...
-    }
-]
 
+NETWORK_IDS = {"mainnet": 0}
+
+SHELLEY_TEST_VECTORS_MNEMONIC = (
+    "test walk nut penalty hip pave soap entry language right filter choice"
+)
+
+class InputAction():
```
Comment (janmazak) [inline tests/device_tests/test_msg_cardano_sign_transaction.py:27]: No parentheses.

**Pull request merged**